agent/consul/internal_endpoint.go

```diff
@@ -122,7 +122,22 @@ func (m *Internal) ServiceDump(args *structs.ServiceDumpRequest, reply *structs.
 				return err
 			}
 
-			reply.Nodes = raw.(structs.CheckServiceNodes)
+			nodes = raw.(structs.CheckServiceNodes)
+			for i, n := range nodes {
+				if n.Service.IsGateway() {
```

Comment:
This would make a call to state.GatewayServices for mesh gateways as well, what if we only checked for ingress/terminating?

Reply:
Ended up refactoring this out and now check for `n.Service.ServiceKindTypical` so that we append gateway services to the service, not the gateway.